examples/ticks_and_spines/tick-formatters.py

```diff
@@ -0,0 +1,93 @@
+"""
+Show the different tick formatters
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.ticker as ticker
+
+
+# Setup a plot such that only the bottom spine is shown
+def setup(ax):
+    ax.spines['right'].set_color('none')
+    ax.spines['left'].set_color('none')
+    ax.yaxis.set_major_locator(ticker.NullLocator())
+    ax.spines['top'].set_color('none')
+    ax.xaxis.set_ticks_position('bottom')
+    ax.tick_params(which='major', width=1.00, length=5)
+    ax.tick_params(which='minor', width=0.75, length=2.5, labelsize=10)
+    ax.set_xlim(0, 5)
+    ax.set_ylim(0, 1)
+    ax.patch.set_alpha(0.0)
+
+
+plt.figure(figsize=(8, 6))
+n = 6
+
+# Null formatter
+ax = plt.subplot(n, 1, 1)
+setup(ax)
+ax.xaxis.set_major_locator(ticker.MultipleLocator(1.00))
+ax.xaxis.set_minor_locator(ticker.MultipleLocator(0.25))
+ax.xaxis.set_major_formatter(ticker.NullFormatter())
+ax.xaxis.set_minor_formatter(ticker.NullFormatter())
+ax.text(0.0, 0.5, "NullFormatter()", fontsize=16, transform=ax.transAxes)
+
+# Fixed formatter
+ax = plt.subplot(n, 1, 2)
+setup(ax)
+ax.xaxis.set_major_locator(ticker.MultipleLocator(1.0))
+ax.xaxis.set_minor_locator(ticker.MultipleLocator(0.25))
+majors = ["", "0", "1", "2", "3", "4", "5"]
+ax.xaxis.set_major_formatter(ticker.FixedFormatter(majors))
+minors = [""] + ["%.2f" % (x-int(x)) if (x-int(x))
+                 else "" for x in np.arange(0, 5, 0.25)]
+ax.xaxis.set_minor_formatter(ticker.FixedFormatter(minors))
+ax.text(0.0, 0.5, "FixedFormatter(['', '0', '1', ...])",
+        fontsize=15, transform=ax.transAxes)
+
+
+# Func formatter
+def major_formatter(x, pos):
+    return "[%.2f]" % x
+
+
+def minor_formatter(x, pos):
+    return "(%.2f)" % x
+
+
+ax = plt.subplot(n, 1, 3)
+setup(ax)
+ax.xaxis.set_major_locator(ticker.MultipleLocator(1.00))
+ax.xaxis.set_minor_locator(ticker.MultipleLocator(0.25))
+ax.xaxis.set_major_formatter(ticker.FuncFormatter(major_formatter))
+ax.text(0.0, 0.5, "FuncFormatter(myfunc)", fontsize=15, transform=ax.transAxes)
```

Comment:
`FuncFormatter(lambda x, pos: "[%.2f]" % x)` seems a better caption?  Also, the `minor_formatter` function seems unused?